### PR TITLE
Permissions frontent improvements

### PR DESCRIPTION
**Pull request by mathjazz**
4 smaller improvements to the permissions page. Details in commit messages.

@jotes r?

### CODE REVIEW

File: pontoon/base/models.py [hunk 1]
```diff
@@ -601,7 +588,6 @@ def create_users_map(qs, keyfield):
                 project_locale['project__name'],
                 projects_all_users[group_pk],
                 projects_translators[group_pk],
-                projects_contributors[project_locale['project__pk']],
```
Comment (mathjazz) [inline pontoon/base/models.py]: Not needed. We always want to use team contributors.

File: pontoon/base/static/js/manage_permissions.js [hunk 2]
```diff
@@ -102,7 +107,7 @@ $(function() {
 
     // Update menu (must be above Copying Translators)
     $(this).addClass('hidden');
-    if ($('#project-selector .menu li:visible').length === 0) {
+    if ($('#project-selector .menu li:not(".hidden")').length === 0) {
```
Comment (mathjazz) [inline pontoon/base/static/js/manage_permissions.js]: The `:visible` selector wasn't reliable - if you searched for a project, items could get hidden, so if only one project matched the criteria and you added it, the selector disappeared.

File: pontoon/base/templates/widgets/multiple_choice.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
     </div>
     <ul>
       {% for user in choices %}
-      <li class="clearfix{% if user in extra %} contributor{% endif %}" data-id="{{ user.id }}" title="{{ user.first_name }}"><span class="arrow fa"></span>{{ user.email }}</li>
+      <li class="clearfix{% if user.email in extra %} contributor{% endif %}" data-id="{{ user.id }}" title="{{ user.first_name }}"><span class="arrow fa"></span>{{ user.email }}</li>
```
Comment (mathjazz) [inline pontoon/base/templates/widgets/multiple_choice.html]: That was actually a bug. For project permissions, we compared a user dict to user objects, so we never had any matches.

File: pontoon/base/views.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def locale_permissions(request, locale):
     translators = l.translators_group.user_set.exclude(pk__in=managers).all()
     all_users = User.objects.exclude(pk__in=managers).exclude(pk__in=translators).exclude(email='')
 
-    contributors = User.translators.filter(translation__locale=l).distinct()
+    contributors = User.translators.filter(translation__locale=l).values_list('email', flat=True).distinct()
```
Comment (mathjazz) [inline pontoon/base/views.py]: Same as above: That was actually a bug. For project permissions, we compared a user dict to user objects, so we never had any matches.

File: pontoon/base/models.py (hunk 1 above)
Comment (jotes) [inline pontoon/base/models.py]: totally makes sense - and that decreases queries count!

File: pontoon/base/static/js/manage_permissions.js (hunk 2 above)
Comment (jotes) [inline pontoon/base/static/js/manage_permissions.js]: I think we're near this point when some kind if UI testing viaselenium or something similar would be helpful.

File: pontoon/base/templates/widgets/multiple_choice.html (hunk 3 above)
Comment (jotes) [inline pontoon/base/templates/widgets/multiple_choice.html]: There's a huge possibility that I broke this during doing my performance refactorings :(

**Pull request merged**